PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -420,47 +428,32 @@ VerifyFwVersion (
 
   This function will get state machine flag from the bootloader reserved region
   First byte in the booloader reserved region is state machine flag
-
-  @param[in, out] StateMachine  Pointer to state machine flag byte.
+  
+  @param[in]      FwUpdStatusOffset Offset to the fw update status structure in reserved region.
+  @param[in, out] StateMachine      Pointer to state machine flag byte.
 
   @retval  EFI_SUCCESS        State machine flag found.
   @retval  others             Error while getting state machine flag.
 **/
 EFI_STATUS
 GetStateMachineFlag (
+  IN     UINT32   FwUpdStatusOffset,
   IN OUT UINT8    *StateMachine
 )
 {
-  EFI_STATUS    Status;
-  UINT32        RsvdBase;
-  UINT32        RsvdOffset;
-  UINT32        RsvdSize;
-  FLASH_MAP     *FlashMap;
-
-  FlashMap = GetFlashMapPtr();
-  if (FlashMap == NULL) {
-    return EFI_NOT_FOUND;
-  }
+  EFI_STATUS              Status;
+  FIRMWARE_UPDATE_STATUS  FwUpdStatus;
 
   //
-  // Get bootloader reserved region base and size
+  // Read from the reserved region and return state machine
   //
-  Status = GetComponentInfoByPartition (FLASH_MAP_SIG_BLRESERVED, FALSE, &RsvdBase, &RsvdSize);
+  Status = BootMediaRead (FwUpdStatusOffset, sizeof(FIRMWARE_UPDATE_STATUS), (UINT8 *)&FwUpdStatus);
```

Comment:
I saw sometimes you use flash map to access the structure.    Why do we need to read media to know the status ?

Reply:
I see issues with reading directly. sometimes I am getting cached structure.
Let me work on that in a later patch

Reply:
OK, understood.  It is OK to do this.  But we will need to be consistent in the whole library.